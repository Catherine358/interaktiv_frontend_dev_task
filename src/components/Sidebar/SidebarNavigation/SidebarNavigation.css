.sidebar-nav {
  margin-bottom: 20px;
}
