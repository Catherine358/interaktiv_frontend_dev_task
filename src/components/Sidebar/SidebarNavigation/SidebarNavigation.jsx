import { SmallCardView, TitleView } from '../../Views';
import { dataProtection, fraudPrevention, itSecurity, wphg } from '../../../assets';

import './SidebarNavigation.css';

const SidebarNavigation = () => (
    <section className="sidebar-nav">
      <TitleView title="Navigation" />
      <SmallCardView dataTestId="navigation-item" title="Money Laundering and Fraud Prevention" image={fraudPrevention} />
      <SmallCardView dataTestId="navigation-item" title="IT security and information security" image={itSecurity} />
      <SmallCardView dataTestId="navigation-item" title="Data Protection" image={dataProtection} />
      <SmallCardView dataTestId="navigation-item" title="WpHG-Compliance" image={wphg} />
    </section>
  );

export default SidebarNavigation;
