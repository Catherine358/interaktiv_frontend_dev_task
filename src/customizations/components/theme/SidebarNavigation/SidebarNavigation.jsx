import TitleView from '../TitleView/TitleView';
import SmallView from '../SmallView/SmallView';
import { dataProtection, fraudPrevention, itSecurity, wphg } from '../../../../assets';

import './SidebarNavigation.css';

const SidebarNavigation = () => (
    <section className="sidebar-nav">
      <TitleView title="Navigation" />
      <SmallView title="Money Laundering and Fraud Prevention" image={fraudPrevention} />
      <SmallView title="IT security and information security" image={itSecurity} />
      <SmallView title="Data Protection" image={dataProtection} />
      <SmallView title="WpHG-Compliance" image={wphg} />
    </section>
  );

export default SidebarNavigation;
