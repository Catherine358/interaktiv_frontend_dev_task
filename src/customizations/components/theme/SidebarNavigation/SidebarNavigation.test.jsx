import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';


import SidebarNavigation from './SidebarNavigation';

describe('SidebarNavigation', () => {
  it('renders sidebar navigation', async () => {
    const component = render(
      <SidebarNavigation  />
    );
    expect(component).toMatchSnapshot();
    expect(screen.getByText(/navigation/i)).toBeInTheDocument();
    const items = await screen.queryAllByTestId('navigation-item');
    expect(items).toHaveLength(4);
  });
});
