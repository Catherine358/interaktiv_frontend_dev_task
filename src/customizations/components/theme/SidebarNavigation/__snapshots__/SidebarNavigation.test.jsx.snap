// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarNavigation renders sidebar navigation 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sidebar-nav"
      >
        <div
          class="title-container"
        >
          <p>
            Navigation
          </p>
        </div>
        <div
          class="ui segment"
          data-testid="navigation-item"
          horizontal="true"
        >
          <div
            class="small-card-img-container"
          >
            <img
              alt="Money Laundering and Fraud Prevention"
              class="ui image"
              height="35"
              src="fraud_prevention.png"
              width="35"
            />
          </div>
          <p>
            Money Laundering and Fraud Prevention
          </p>
        </div>
        <div
          class="ui segment"
          data-testid="navigation-item"
          horizontal="true"
        >
          <div
            class="small-card-img-container"
          >
            <img
              alt="IT security and information security"
              class="ui image"
              height="35"
              src="it_security.png"
              width="35"
            />
          </div>
          <p>
            IT security and information security
          </p>
        </div>
        <div
          class="ui segment"
          data-testid="navigation-item"
          horizontal="true"
        >
          <div
            class="small-card-img-container"
          >
            <img
              alt="Data Protection"
              class="ui image"
              height="35"
              src="data_protection.png"
              width="35"
            />
          </div>
          <p>
            Data Protection
          </p>
        </div>
        <div
          class="ui segment"
          data-testid="navigation-item"
          horizontal="true"
        >
          <div
            class="small-card-img-container"
          >
            <img
              alt="WpHG-Compliance"
              class="ui image"
              height="35"
              src="wphg.png"
              width="35"
            />
          </div>
          <p>
            WpHG-Compliance
          </p>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sidebar-nav"
    >
      <div
        class="title-container"
      >
        <p>
          Navigation
        </p>
      </div>
      <div
        class="ui segment"
        data-testid="navigation-item"
        horizontal="true"
      >
        <div
          class="small-card-img-container"
        >
          <img
            alt="Money Laundering and Fraud Prevention"
            class="ui image"
            height="35"
            src="fraud_prevention.png"
            width="35"
          />
        </div>
        <p>
          Money Laundering and Fraud Prevention
        </p>
      </div>
      <div
        class="ui segment"
        data-testid="navigation-item"
        horizontal="true"
      >
        <div
          class="small-card-img-container"
        >
          <img
            alt="IT security and information security"
            class="ui image"
            height="35"
            src="it_security.png"
            width="35"
          />
        </div>
        <p>
          IT security and information security
        </p>
      </div>
      <div
        class="ui segment"
        data-testid="navigation-item"
        horizontal="true"
      >
        <div
          class="small-card-img-container"
        >
          <img
            alt="Data Protection"
            class="ui image"
            height="35"
            src="data_protection.png"
            width="35"
          />
        </div>
        <p>
          Data Protection
        </p>
      </div>
      <div
        class="ui segment"
        data-testid="navigation-item"
        horizontal="true"
      >
        <div
          class="small-card-img-container"
        >
          <img
            alt="WpHG-Compliance"
            class="ui image"
            height="35"
            src="wphg.png"
            width="35"
          />
        </div>
        <p>
          WpHG-Compliance
        </p>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
